cmd/agent/workspace/up.go

```diff
@@ -206,13 +210,19 @@ func initWorkspace(ctx context.Context, cancel context.CancelFunc, workspaceInfo
 	}
 
 	// install docker in background
-	errChan := make(chan error, 1)
+	errChan := make(chan error, 2)
 	go func() {
 		if !workspaceInfo.Agent.IsDockerDriver() || workspaceInfo.Agent.Docker.Install == "false" {
 			errChan <- nil
 		} else {
 			errChan <- installDocker(logger)
 		}
+		// If we are provisioning the machine, ensure the daemon has required options
+		if workspaceInfo.Machine != nil {
```

Comment:
is this the best way to check if we are using a machine provider? I only want to configure the daemon on infrastructure we provision

Reply:
You could check for docker driver and if we're on a remote machine:
`workspaceInfo.Agent.IsDockerDriver()` and `workspaceInfo.Agent.Local.Bool()`